framework/components/ecl-buttons/ecl-buttons.twig

```diff
@@ -0,0 +1,19 @@
+{% set type = type|default('button') %}
+{% set modifier = modifier|default('default') %}
+{% set extraClass = '' %}
+
+{% if isBlock is defined and isBlock == true %}
+  {% set extraClass = ' ecl-button--block' %}
+{% endif %}
+
+{% if icon is defined %}
+  {% set extraClass = extraClass ~ ' ecl-button--' ~ icon %}
```

Comment:
Adding a comment on this one will be appreciated, it currently seems a bit magical and takes time to understand.